extensions-contrib/compressed-bigdecimal/src/main/java/org/apache/druid/compressedbigdecimal/CompressedBigDecimal.java

```diff
@@ -274,13 +276,74 @@ protected static <S> int signumInternal(int size, S rhs, ToIntBiFunction<S, Inte
   /* (non-Javadoc)
    * @see java.lang.Comparable#compareTo(java.lang.Object)
    */
+
   @Override
   public int compareTo(CompressedBigDecimal o)
+  {
+    return compareTo(o, false);
+  }
+  public int compareTo(CompressedBigDecimal o, boolean expectOptimized)
```

Comment:
I notice that this method is called by `compareTo` method at line 283, and a `false` is passed to this method as 2nd parameter. I don't find any other places that call this method with a `true` value.

Reply:
it's called in the unit test. It was a suggestion by @cheddar as a way to detect a regression in the optimized path of compareTo() in unit tests.

Reply:
if it's called by the UT only, usually we declare its visibility as package level, and use `VisibileForTesting` annotation to make it more intuitive.

Reply:
Heh, he's not doing it that way because I hate the `VisibleForTesting` annotation and adjusting the visibility only creates friction for the future without actually protecting anything.  Those things combine to just pretend like the method is not part of the actual contract of the object when it is.  

In this case, there is basically a "strict" version that expects that objects have already been aligned so that we can use a more optimal code path.  But there's a more loose version as well that just "does the right thing".  We have the tests that exercise this via queries only use the "strict" version because the normal query pipeline should be aligning things so that the more optimal code path is followed, but in production the non-strict version is actually used just in case.  I.e. the tests are there to catch against regressions when running through queries, but there are other tests that also leverage the whole thing.  In all, the "contract" of the object requires us to have both, otherwise we cannot test it.  There also might be a case in production where we want to blow up if there is a mis-alignment, in that case, this method is a great way to achieve that.